### PR TITLE
[THINGY-1] Use mongo in Docker

### PR DESCRIPTION
**Pull request by TripathyM**
https://trello.com/c/sfOrpFwC/1-set-up-mongo-with-docker

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,7 @@
+FROM openjdk:12
```
Comment (TripathyM) [inline Dockerfile:1]: @sankita15 What is the java version that we agreed on? Is Java 12 ok?

**Review by TripathyM — COMMENTED** [review]
I did not add tests 🙈. I will do soon.

**Review by TripathyM — COMMENTED** [review]
Test review to check slack integration

File: Dockerfile (hunk 1 above)
Comment (sankita15tw) [reply Dockerfile:1]: Its java 11

File: src/main/java/com/thingy/model/Todo.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import lombok.NoArgsConstructor;
+import org.bson.types.ObjectId;
+import org.springframework.data.annotation.Id;
+
+@AllArgsConstructor
+@NoArgsConstructor
+@Data
+public class Todo {
+  @Id
+  private ObjectId _id;
```
Comment (sankita15tw) [inline src/main/java/com/thingy/model/Todo.java]: Why _id I guess we can give simply id as well

File: src/main/resources/application.yaml [hunk 3]
```diff
@@ -1,6 +1,6 @@
 spring:
   data:
     mongodb:
-      host: localhost
+      host: mymongo
```
Comment (sankita15tw) [inline src/main/resources/application.yaml]: didn't get this? why we change this?

File: src/main/java/com/thingy/model/Todo.java (hunk 2 above)
Comment (TripathyM) [reply src/main/java/com/thingy/model/Todo.java]: Very nice question. I see that _id is a mandatory field for all documents in MongoDB. But I can still name it as `id` in the java class. Refer `How the '_id' field is handled in the mapping layer
` section in https://docs.spring.io/spring-data/mongodb/docs/1.3.3.RELEASE/reference/html/mapping-chapter.html

File: src/main/resources/application.yaml (hunk 3 above)
Comment (TripathyM) [reply src/main/resources/application.yaml]: We are using Docker to spin up a mongo container. Refer to the `docker-compose.yml` file. The hostname of the mongo container is listed there.
But I see your point. These can be configured as environment variables and used appropriately to minimize confusion and dependency.

**Comment by TripathyM** [comment]
> 12 or 11?
> JDK 12 release schedule
> 12/13/2019 Rampdown Phase One (fork from mainline)
> 01/17/2019 Rampdown Phase Two
> 02/07/2019 Release-Candidate Phase
> 03/19/2019 General Availability
> 
> I guess we need to change to 11 then we can merge this PR

I see your point here, but given that we are moving ahead in this project from a learning perspective as well, can we stay with Java 12 until we really see some breaking changes.

**Review by sankita15 — APPROVED** [review]

**Pull request merged**